src/lenses/MarketLens.sol

```diff
@@ -65,26 +65,34 @@ contract MarketLens {
         mimInBentoBox = bentoBox.toAmount(mim, poolBalance, false);
     }
 
-    function getMaxBorrowForCauldronV2User(ICauldronV2 cauldron) public view returns (uint256) {
+    function getMaxMarketBorrowForCauldronV2(ICauldronV2 cauldron) public view returns (uint256) {
         return getMimInBentoBox(cauldron);
     }
 
-    function getBorrowLimitForCauldronV3User(ICauldronV3 cauldron) private view returns (uint256) {
-        (uint256 totalLimit, uint256 borrowPartPerAddress) = cauldron.borrowLimit();
-        return MathLib.min(totalLimit, borrowPartPerAddress);
+    function getMaxUserBorrowForCauldronV2(ICauldronV2 cauldron) public view returns (uint256) {
+        return getMimInBentoBox(cauldron);
     }
 
     // Returns the maximum amount that can be borrowed across all users
-    function getMaxBorrowForCauldronV3Market(ICauldronV3 cauldron) public view returns (uint256) {
-        (uint256 totalLimit, ) = cauldron.borrowLimit();
-        return MathLib.min(totalLimit, getMimInBentoBox(cauldron));
+    function getMaxMarketBorrowForCauldronV3(ICauldronV3 cauldron) public view returns (uint256) {
+        (uint256 totalBorrowLimit, ) = cauldron.borrowLimit();
+
+        uint256 mimInBentoBox = getMimInBentoBox(cauldron);
+        uint256 remainingBorrowLimit = MathLib.subWithZeroFloor(totalBorrowLimit, getTotalBorrowed(cauldron));
+
+        return MathLib.min(mimInBentoBox, remainingBorrowLimit);
     }
 
     // Returns the maximum amount that a single user can borrow
-    function getMaxBorrowForCauldronV3User(ICauldronV3 cauldron) public view returns (uint256) {
-        uint256 mimInBentoBox = getMimInBentoBox(cauldron);
-        uint256 userBorrowLimit = getBorrowLimitForCauldronV3User(cauldron);
-        return MathLib.min(userBorrowLimit, mimInBentoBox);
+    function getMaxUserBorrowForCauldronV3(ICauldronV3 cauldron) public view returns (uint256) {
+        (uint256 totalBorrowLimit, uint256 userBorrowLimit) = cauldron.borrowLimit();
+
+        uint256[] memory values = new uint256[](3);
```

Comment:
Here it's a and there's no gas implication really but in general, you want to avoid creating such array and looping.  I understand that in trad programming this is cleaner but here I suggest doing:

```
return MathLib.min(
            MathLib.min(getMimInBentoBox(cauldron), MathLib.subWithZeroFloor(totalBorrowLimit, getTotalBorrowed(cauldron))),
            userBorrowLimit
        );
```

Reply:
42266 gas versus 41533 without the loop